qemu/tests/cpu_device_hotpluggable.py

```diff
@@ -0,0 +1,127 @@
+                                       verify_wait_timeout):
+                logging.error("CPU quantity mismatched! Guest got %s but the "
+                              "expected is %s", vm.get_cpu_count(), smp)
+                test.fail(mismatch_text)
+            for cpu_id in cpu_ids:
+                session.cmd(cmd % (1, cpu_id))
+        except ShellCmdError as err:
+            logging.error(str(err))
+            test.error("Failed to change the CPU state on guest.")
+
+    login_timeout = params.get_numeric("login_timeout", 360)
+    os_type = params["os_type"]
+    hotpluggable_test = params["hotpluggable_test"]
+    verify_wait_timeout = params.get_numeric("verify_wait_timeout", 60)
+    sub_test_type = params.get("sub_test_type")
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=login_timeout)
+    smp = vm.cpuinfo.smp
+    vcpus_count = vm.params.get_numeric("vcpus_count", 1)
+    vcpu_devices = params.objects("vcpu_devices")
+    guest_cpu_ids = cpu_utils.get_guest_cpu_ids(session, os_type)
+
+    if hotpluggable_test == "hotplug":
+        pluggable_vcpu_dev = vcpu_devices
+        pluggable_vcpu = vcpus_count * len(pluggable_vcpu_dev)
+    else:
+        pluggable_vcpu_dev = vcpu_devices[::-1]
+        pluggable_vcpu = -(vcpus_count * len(pluggable_vcpu_dev))
+    expected_vcpus = vm.get_cpu_count() + pluggable_vcpu
+    if params.get("pause_vm_before_hotplug", "no") == "yes":
+        error_context.context("Pause guest before %s" % hotpluggable_test,
+                              logging.info)
+        vm.pause()
+    error_context.context("%s all vcpu devices" % hotpluggable_test,
+                          logging.info)
+    for vcpu_dev in pluggable_vcpu_dev:
+        getattr(vm, "%s_vcpu_device" % hotpluggable_test)(vcpu_dev)
+    if vm.is_paused():
+        error_context.context("Resume guest after %s" % hotpluggable_test,
+                              logging.info)
+        vm.resume()
+
+    mismatch_text = "Actual number of guest CPUs is not equal to the expected."
+    if not utils_misc.wait_for(lambda: vm.get_cpu_count() == expected_vcpus,
+                               verify_wait_timeout):
+        logging.error("CPU quantity mismatched! Guest got %s but the expected"
+                      " is %s", vm.get_cpu_count(), expected_vcpus)
+        test.fail(mismatch_text)
+    current_guest_cpu_ids = cpu_utils.get_guest_cpu_ids(session, os_type)
+
+    if sub_test_type:
+        eval("sub_%s()" % sub_test_type)
+
+    if vm.is_alive() and not utils_misc.wait_for(
+            lambda: vm.get_cpu_count() == expected_vcpus, verify_wait_timeout):
+        test.fail(mismatch_text)
+        if not cpu_utils.check_guest_cpu_topology(session, os_type,
+                                                  vm.cpuinfo):
+            test.fail("CPU topology of guest is inconsistent with "
+                      "expectations.")
+    if session:
```

Comment:
Let's check guest dmesg before the end.

Reply:
`verify_guest_dmesg = yes` is the default setting